app/src/main/java/org/android/go/sopt/ui/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,107 @@
+import org.android.go.sopt.ui.login.LoginActivity
+import org.android.go.sopt.util.*
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignUpBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.root.setOnClickListener {
+            hideKeyboard()
+        }
+
+        binding.etId.addTextChangedListener {
+            if (!checkLengthOfId(it.toString())) {
+                binding.tvIdLimitError.visibility = View.VISIBLE
+            } else {
+                binding.tvIdLimitError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etPw.addTextChangedListener {
+            if (!checkLengthOfPw(it.toString())) {
+                binding.tvPwLimitError.visibility = View.VISIBLE
+            } else {
+                binding.tvPwLimitError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etName.addTextChangedListener {
+            if (it.toString().isEmpty()) {
+                binding.tvNameEmptyError.visibility = View.VISIBLE
+            } else {
+                binding.tvNameEmptyError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etHobby.addTextChangedListener {
+            if (it.toString().isEmpty()) {
+                binding.tvHobbyEmptyError.visibility = View.VISIBLE
+            } else {
+                binding.tvHobbyEmptyError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.btnSignUp.setOnClickListener {
+            val inputs = getAllInputValues()
+            if(checkInputValidity(inputs)){
+                val intent = Intent(this, LoginActivity::class.java).apply {
+                    putExtra(ID_KEY, inputs[0])
+                    putExtra(PW_KEY, inputs[1])
+                    putExtra(NAME_KEY, inputs[2])
+                    putExtra(HOBBY_KEY, inputs[3])
+                }
+                setResult(RESULT_OK, intent)
+                finish()
+            }else{
+                Toast.makeText(this, INVALID_INPUT_ERROR, Toast.LENGTH_SHORT).show()
+            }
+        }
+    }
+
+    private fun hideKeyboard() {
+        val imm = getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(binding.root.windowToken, 0)
+    }
+
+    private fun checkInputValidity(inputs: List<String>): Boolean {
+        return checkLengthOfId(inputs[0]) && checkLengthOfPw(inputs[1]) && inputs[2].isNotEmpty() && inputs[3].isNotEmpty()
+    }
+
+    private fun getAllInputValues(): List<String> {
+        val id = binding.etId.text.toString()
+        val pw = binding.etPw.text.toString()
+        val name = binding.etName.text.toString()
+        val hobby = binding.etHobby.text.toString()
+        return listOf(id, pw, name, hobby)
+    }
```

Comment:
```suggestion
    private fun getAllInputValues(): User {
        val id = binding.etId.text.toString()
        val pw = binding.etPw.text.toString()
        val name = binding.etName.text.toString()
        val hobby = binding.etHobby.text.toString()
        return User(id, pw, name, hobby)
    }
```
Data Class로 정의하신 User 클래스를 활용해보는 건 어떨까요?